tasks/test_results_processor.py

```diff
@@ -0,0 +1,200 @@
+
+        if self.should_delete_archive(commit_yaml):
+            repository = (
+                db_session.query(Repository)
+                .filter(Repository.repoid == int(repoid))
+                .first()
+            )
+            self.delete_archive(
+                commitid, repository, commit_yaml, uploads_to_delete=upload_list
+            )
+
+        return {
+            "successful": True,
+            "testrun_list": [self.testrun_to_dict(t) for t in testrun_list],
+        }
+
+    def save_report(
+        self, db_session: Session, testrun_list: List[Testrun], upload: Upload
+    ):
+        repo_tests = (
+            db_session.query(Test).filter_by(repoid=upload.report.commit.repoid).all()
+        )
+        test_set = set()
+        for test in repo_tests:
+            test_set.add(f"{test.testsuite}::{test.name}")
+        for testrun in testrun_list:
+            if f"{testrun.testsuite}::{testrun.name}" not in test_set:
+                test = Test(
+                    repoid=upload.report.commit.repoid,
+                    name=testrun.name,
+                    testsuite=testrun.testsuite,
+                )
+                db_session.add(test)
+                db_session.flush()
+
+            db_session.add(
+                TestInstance(
+                    test_id=test.id,
+                    duration_seconds=testrun.duration,
+                    outcome=int(testrun.outcome),
+                    upload_id=upload.id,
+                )
+            )
+            db_session.flush()
+
+    def process_individual_arg(self, upload: Upload, repository) -> List[Testrun]:
+        archive_service = ArchiveService(repository)
+
+        payload_bytes = archive_service.read_file(upload.storage_path)
+        data = json.loads(payload_bytes)
+
+        testrun_list = []
+
+        for file in data["test_results_files"]:
+            file = file["data"]
+            print(file)
+            file_bytes = BytesIO(zlib.decompress(base64.b64decode(file)))
+            print(file_bytes)
+
+            first_line = file_bytes.readline()
+            file_bytes.seek(0)
+
+            file_content = file_bytes.read()
+
+            # TODO: improve report matching capabilities
+            # use file extensions?
+            # maybe do the matching in the testing result parser lib?
+            first_line = bytes("".join(first_line.decode("utf-8").split()), "utf-8")
+            try:
+                if first_line.startswith(b"<?xml"):
+                    testrun_list = parse_junit_xml(file_content)
+                elif first_line.startswith(b'{"pytest_version": "7.4.3'):
+                    testrun_list = parse_pytest_reportlog(file_content)
+                elif first_line.startswith(b'"{"numTotalTestSuites":'):
+                    testrun_list = parse_vitest_json(file_content)
+            except Exception as e:
+                log.warning(f"Error parsing: {file_content.decode()}")
```

Comment:
Please use a static message an add extra args to the logger. That will make it easier to search for logs and it's the norm for worker logs